From b2fdbeefbc9918b3092c53e8d77d47ba19b2df4b Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Wed, 16 Mar 2022 19:47:53 +0800
Subject: [PATCH] Fix Openssl 3.0 error message for OSSL_PROVIDER_load 'legacy' failed.

---
 src/Platform.cc |   1 +
 1 files changed, 1 insertion(+)

diff --git a/src/Platform.cc b/src/Platform.cc
index 9f3ddb25..cb45b311 100644
--- a/src/Platform.cc
+++ b/src/Platform.cc
@@ -43,6 +43,7 @@

 #ifdef HAVE_OPENSSL
 #  include <openssl/err.h>
+#  include <openssl/provider.h>
 #  include <openssl/ssl.h>
 #  include "libssl_compat.h"
 #endif // HAVE_OPENSSL
--
2.35.1
