From df4a9fa881c5ac1264b918dec2b689e961655916 Mon Sep 17 00:00:00 2001
From: Zhijie He <hezhijie0327@hotmail.com>
Date: Mon, 26 Jul 2021 08:14:56 +0800
Subject: [PATCH] Add random IP address selection support

---
 src/AbstractCommand.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/AbstractCommand.cc b/src/AbstractCommand.cc
index 1502d570..b8514242 100644
--- a/src/AbstractCommand.cc
+++ b/src/AbstractCommand.cc
@@ -767,7 +767,7 @@ std::string AbstractCommand::resolveHostname(std::vector<std::string>& addrs,
 
   e_->findAllCachedIPAddresses(std::back_inserter(addrs), hostname, port);
   if (!addrs.empty()) {
-    auto ipaddr = addrs.front();
+    auto ipaddr = addrs[rand() % addrs.size()];
     A2_LOG_INFO(fmt(MSG_DNS_CACHE_HIT, getCuid(), hostname.c_str(),
                     strjoin(std::begin(addrs), std::end(addrs), ", ").c_str()));
     return ipaddr;
-- 
2.32.0

